test/epoch_token_locker.js

```diff
@@ -181,63 +181,63 @@ contract("EpochTokenLocker", async (accounts) => {
-      await epochTokenLocker.subtractBalanceTest(user_2, ERC20.address, 50)
-
-      assert.equal(await epochTokenLocker.getBalance(user_2, ERC20.address), 50)
-    })
-    it("throws in case of underflow", async () => {
-      const epochTokenLocker = await EpochTokenLockerTestInterface.new()
-      const ERC20 = await MockContract.new()
-
-      await truffleAssert.reverts(epochTokenLocker.subtractBalanceTest(user_1, ERC20.address, 50))
-    })
-  })
+  // describe("addBalance", () => {
+  //   it("modifies the balance by adding", async () => {
+  //     const epochTokenLocker = await EpochTokenLockerTestInterface.new()
+  //     const ERC20 = await MockContract.new()
+
+  //     await epochTokenLocker.addBalanceTest(user_1, ERC20.address, 100)
+
+  //     assert.equal(await epochTokenLocker.getBalance(user_1, ERC20.address), 100)
+  //   })
+  // })
+  // describe("addBalanceAndPostponeWithdraw", () => {
+  //   it("does not postpone a withdrawRequest for a future epoch", async () => {
+  //     const epochTokenLocker = await EpochTokenLockerTestInterface.new()
+  //     const ERC20 = await MockContract.new()
+  //     const currentStateIndex = await epochTokenLocker.getCurrentStateIndex.call()
+  //     await epochTokenLocker.requestWithdraw(ERC20.address, 100, { from: user_1 })
+  //     await epochTokenLocker.addBalanceAndPostponeWithdrawTest(user_1, ERC20.address, 100)
+
+  //     assert.equal((await epochTokenLocker.getPendingWithdrawBatchNumber(user_1, ERC20.address)).toNumber(), currentStateIndex, "State index updated incorrectly during call to addBalanceAndPostponeWithdrawTest")
+  //   })
+  //   it("postpones a withdrawRequest for a current epoch", async () => {
+  //     const epochTokenLocker = await EpochTokenLockerTestInterface.new()
+  //     const ERC20 = await MockContract.new()
+  //     const currentStateIndex = (await epochTokenLocker.getCurrentStateIndex.call()).toNumber()
+  //     await epochTokenLocker.requestWithdraw(ERC20.address, 100, { from: user_1 })
+  //     await waitForNSeconds(BATCH_TIME + 1)
+  //     await epochTokenLocker.addBalanceAndPostponeWithdrawTest(user_1, ERC20.address, 100)
+
+  //     assert.equal((await epochTokenLocker.getPendingWithdrawBatchNumber(user_1, ERC20.address)).toNumber(), currentStateIndex + 1, "State index not updated during call to addBalanceAndPostponeWithdrawTest")
+  //   })
+  // })
+  // describe("subtractBalance", () => {
+  //   it("modifies the balance by subtraction", async () => {
+  //     const epochTokenLocker = await EpochTokenLockerTestInterface.new()
+  //     const ERC20 = await MockContract.new()
+
+  //     await epochTokenLocker.addBalanceTest(user_1, ERC20.address, 100)
+  //     await epochTokenLocker.subtractBalanceTest(user_1, ERC20.address, 50)
+
+  //     assert.equal(await epochTokenLocker.getBalance(user_1, ERC20.address), 50)
+  //   })
+  //   it("modifies the balance by subtracting on behalf of someone else", async () => {
```

Comment:
@josojo. This test does not appear to do what it says. User 2 adds balance and then subtracts their own. Perhaps I am mistaken, could you please elaborate on this.

Reply:
Furthermore, I do not see any particular reason this function would require such a test (since it contains no requirements checking the relevant user)

https://github.com/gnosis/dex-contracts/blob/28220de3a116784d62941778ae78a1bcca2be1f7/contracts/EpochTokenLocker.sol#L139-L142